src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,216 @@
+            int order = i + 1;
+            if (task.isDone()) {
+                doneCount++;
+            }
+            System.out.println(String.format("\t%d. %s", order, task.toString()));
+        }
+        System.out.println(String.format("You have %d tasks and you completed %d of them", listCount, doneCount));
+    }
+
+
+    /**
+     * Adds new task to the list
+     *
+     * @param type Type of the new task
+     * @param args Arguments that are passed with the type
+     */
+    public static void addTask(Task.Type type, String args) {
+        try {
+            switch (type) {
+            case DEADLINE:
+                tasks[listCount] = createDeadline(args);
+                break;
+            case TODO:
+                tasks[listCount] = createTodo(args);
+                break;
+            case EVENT:
+                tasks[listCount] = createEvent(args);
+                break;
+            }
+            System.out.println("Okey Dokey! Added: " + tasks[listCount++]);
+        } catch (AddDeadlineException ade) {
+            System.out.println("Oh, no! Deadline's /by cannot be empty!");
+        } catch (AddEventException aee) {
+            System.out.println("Oh, no! Event's /at cannot be empty!");
+        } catch (EmptyNameException ene) {
+            System.out.println("Oh, no! Task cannot be created without a name!");
+        } catch (NullPointerException npe) {
+            System.out.println("Oh, no! More arguments is needed!");
+        }
+    }
+
+    /**
+     * Creates a new deadline task
+     *
+     * @param args Arguments of the command
+     * @return Newly created deadline
+     */
+    public static Deadline createDeadline(String args) throws EmptyNameException, AddDeadlineException {
+        String[] argArr = args.split(" /by ");
+        String name;
+        String by;
+        name = argArr[0].trim();
+        if (name.equals("")) {
+            throw new EmptyNameException();
+        }
+        if (argArr.length != 2) {
+            throw new AddDeadlineException();
+        }
+        by = argArr[1].trim();
+        if (by.equals("")) {
+            throw new AddDeadlineException();
+        }
+        return new Deadline(name, by);
+    }
+
+    /**
+     * Creates a new to-do task
+     *
+     * @param name Name of the to-do task
+     * @return Newly created to-do
+     */
+    public static Todo createTodo(String name) throws EmptyNameException {
+        name = name.trim();
+        if (name.equals("")) {
+            throw new EmptyNameException();
+        }
+        return new Todo(name);
+    }
+
+    /**
+     * Creates a new event task
+     *
+     * @param args Arguments of the command
+     * @return Newly created event
+     */
+    public static Event createEvent(String args) throws EmptyNameException, AddEventException {
+        String[] argArr = args.split(" /at ");
+        String name;
+        String at;
+        name = argArr[0].trim();
```

Comment:
Some magic numbers here. Not sure if its better for you to change it

Reply:
I think it's pretty direct because I assign it to name. I asked prof already and he seems cool with it. Thanks for the comment anyway!